### ISSUE TITLE
Feature Request: Mobile clients for Android & iOS

### ISSUE THREAD

**Issue by tswast** [beakerbrowser/beaker#466]
I'd love to be able to run Beaker on an Android or iOS device. I think the offline capabilities would be perfect for a mobile device.

**Comment by pfrazee** [comment]
We probably won't be able to hit this in the near future, but we agree it'd be pretty great

**Comment by quintolet** [comment]
How tightly coupled is Beaker to Electron? There are a handful of browser implementation for mobile devices, I wonder if it'll be worthwhile to do a fork of those. 

A mobile app would instantly win over users for Beaker. Zeronet does this ugly `http://127.0.0.1:43110` thing, but at least that was easier to make it run on mobile.

**Comment by pfrazee** [comment]
Pretty tightly, but here's some good news: a team in our community is working on the problem with a sister project https://github.com/bunsenbrowser

**Comment by RangerMauve** [comment]
[Nodejs-mobile](https://github.com/janeasystems/nodejs-mobile) is a thing now that lets you run Node with all it's fancy features on mobile.

**Comment by L8D** [comment]
I will be hacking on nodejs-mobile to see if there is an easy way to start a PoC beaker browser port.

My goal is to enable users to click dat:// urls and successfully open content in their browser. To solve this problem practically, I'm going to serve dat content over HTTP, and then build a small React Native app that can open dat:// urls by redirecting to the HTTP proxy/bridge.

**Comment by RangerMauve** [comment]
Hey, I was actually about to start doing something with react-native and dat. My goal to start is to get discovery-swarm to work in the react-native JS thread before getting into nodejs-mobile. I'll probably be starting Thursday or Friday.

**Comment by RangerMauve** [comment]
I think it'd be cool if we both worked on this in parallel projects so that there'd be more options in the ecosystem. 🤩

**Comment by luandro** [comment]
I helped port SSB to nodejs-mobile, so got to learn a bit about it.

Ping me if you hit any blockers @RangerMauve and @L8D. Looking forward to see your progress on this.

**Comment by RangerMauve** [comment]
I was actually taking a different approach. I'm running everything within the react-native thread. So far it seems to be a little slow. 😅

You can track the progress [here](https://github.com/RangerMauve/datmobile/releases)

**Comment by Slender1808** [comment]
is there a browser for android that works with hyper protocol ?

**Comment by RangerMauve** [comment]
Yup! There's Gateway Browser on Android and iOS: https://gitlab.com/gateway-browser/gateway
